lib/types/coreNote.js

```diff
@@ -0,0 +1,111 @@
+/*!
+ * Tozny e3db
+ *
+ * LICENSE
+ *
+ * Tozny dual licenses this product. For commercial use, please contact
+ * info@tozny.com. For non-commercial use, this license permits use of the
+ * software only by government agencies, schools, universities, non-profit
+ * organizations or individuals on projects that do not receive external
+ * funding other than government research grants and contracts. Any other use
+ * requires a commercial license. For the full license, please see LICENSE.md,
+ * in this source repository.
+ *
+ * @copyright Copyright (c) 2018-19 Tozny, LLC (https://tozny.com)
+ */
+
+'use strict'
+
+import Serializable from './serializable'
+import { write } from 'fs'
+
+/*
+ * CoreNote represents required note information that is signed before encrypting
+ */
+export default class CoreNote extends Serializable {
```

Comment:
I don't like CoreNote because `Core` doesn't give me context to what this object is for, without reading the comments to see that CoreNote is a collection of the note's message and encryption/signing info. But I don't have a better idea so ignore me.

Reply:
`UnsignedNote` maybe?

Reply:
I'm sure there could be a better name, but here's my justification CoreNote to me represents all the valid information required to be sent off to TozStore; hence the 'core requirements of a note'. It takes away all the server populated values (created at, expiration) and the optional values (max views, etc). This is the document/object that is signed and later validated when a user downloads the note.

Reply:
UnsignedNote, seems a little inaccurate to me because this object will never contain the signature or any other Note information, but if it's clearer I'm happy to change it

Reply:
How do record objects deal with this?

Reply:
After reviewing the note I Suggest keeping parity with records for structure and management of signatures: [RecordInfo](https://github.com/tozny/js-sdk-client-interface/blob/master/lib/types/recordInfo.js) extends the [SignedDocument](https://github.com/tozny/js-sdk-client-interface/blob/master/lib/types/signedDocument.js) object.

It accepts the data object and a standard javascript object as 'clientMeta'. That standard object is copied over into another object. It is compatible with the [Meta](https://github.com/tozny/js-sdk-client-interface/blob/master/lib/types/meta.js) object. You can see this Info object being created and then signed for [local encrypted records in that method](https://github.com/tozny/js-sdk-client-interface/blob/master/lib/client.js#L661). Decryption similarly is [available as well](https://github.com/tozny/js-sdk-client-interface/blob/master/lib/client.js#L706).

As I have explored this, I am understanding CoreNote more, and it does make sense as written.

'CoreNote' could be 'NoteInfo' similar to 'RecordInfo'.

If we want to structure the objects a little more similarly to how Records are structured, we could potentially explore this:

The data contained in the core note: you've got the data, options and meta, and then the crypto information (public keys, eak, mode). Type, PlainMeta, and FileMeta could be moved to the options object since at the end of the day they are optional. A 'NoteKeys' object or similar could house a serializable set of the keys.

```
NoteInfo(Data, NoteKeys, Options)
Note(Data, NokeKeys, Options, Signature)
```

The main question is how we want to structure our data objects. The current implementation is a fine take on it, and matches more closely with the current notes API. I do think we should consider managing the signable and signed notes object separately to match records. As written a user of a `Note` object, would need to constantly use `note.core.field` to access the majority of data. It means a little bit of extra code for us, but a much better experience for users.